src/nvim/message.c

```diff
@@ -270,32 +270,16 @@ void trunc_string(char_u *s, char_u *buf, int room, int buflen)
   }
 
   /* Last part: End of the string. */
-  i = e;
-  if (enc_dbcs != 0) {
-    /* For DBCS going backwards in a string is slow, but
-     * computing the cell width isn't too slow: go forward
-     * until the rest fits. */
-    n = vim_strsize(s + i);
-    while (len + n > room) {
-      n -= ptr2cells(s + i);
-      i += (*mb_ptr2len)(s + i);
-    }
-  } else if (enc_utf8) {
-    /* For UTF-8 we can go backwards easily. */
```

Comment:
I considered it, but decided not to, as from the other comment (` For DBCS going backwards in a string is slow[...]`), I understood that backwards is sort of the default direction. The comment you want to keep is more of a contrast to the previous comment; it is not really an independent comment imo. If we want to keep it, I think we should change it to something like 'iterate backwards over the strings; this is reasonable fast for UTF-8'.